Controller/Api/ApiController.php

```diff
@@ -42,4 +43,29 @@ public function initialize(FilterControllerEvent $event)
 
         parent::initialize($event);
     }
+
+    /**
+     * Creates a new entity.
+     *
+     * @return Response
+     */
+    public function newEntityAction()
+    {
+        $parameters = $this->request->request->all();
+        // handle reason code
+        if (!isset($parameters['reason']) || empty($parameters['reason'])) {
+            $parameters['reason'] = 3;
```

Comment:
do we have constants for these reason codes? I only ask cause I don't know what reason code 3 means...